### PR TITLE
Explain globals input (1-22)

### PR DESCRIPTION
**Pull request by philtweir**
Adds a brief explanation around the parameters section to the documentation, explaining how context is handled.

### CODE REVIEW

File: docs/workflows.md [hunk 1]
```diff
@@ -106,6 +106,11 @@ and treat them as parameters. It will try to get the type from the typehint, or
 the value that you have set it to. This only works for basic types (and dict/lists of
 those).
 
+While global variables are implicit input to the Python function **note that**:
+
+1. in CWL, they will be rendered as explicit global input to a step
+2. as input, they are read-only, and must not be updated
```
Comment (elleryames) [inline docs/workflows.md:112]: Is this enforced ("cannot be updated"), or is a user responsible for good behavior ("must not")? Could for example, a user do 

```python
INPUT_NUM = 3
@task()
def rotate(num: int) -> int:
    """Rotate an integer."""
    return INPUT_NUM += num
```

Comment (philtweir) [reply docs/workflows.md:112]: You're right, it's not enforced - in practice, changing INPUT_NUM would be like changing a function argument, so would be allowed but not have any visibility outside the function. There is some discussion around this point at #20 but I am not too sure how best we could restrict it. Simple cases, like assignment, we might be able to do by inspection, but something like:

```python
INPUT_PARAMETERS = {'my_param': {}}
@task()
def rotate(num: int) -> int:
    my_param = INPUT_PARAMETERS['my_param']
    my_param['my_subparam'] = 1
```

would be hard to spot.

Comment (philtweir) [reply docs/workflows.md:112]: (also, bear in mind, we only inspect non-nested tasks, we don't execute them, so we do not have an opportunity to compare at start/end, or something like that)

Comment (elleryames) [reply docs/workflows.md:112]: The example I gave leads to a `UnboundLocalError` while your modification of the global sails through undetected. Either way the user is going to find out (or not) at run time.

Comment (elleryames) [reply docs/workflows.md:112]: The current behavior is fine I think, but we might emphasize the proper use a bit more heavily. Perhaps something along the lines of 

```
**Important:** global parameters should be treated as **read-only** and must not be updated. 

While global variables are implicit input to the Python function **note that** in CWL, they will be rendered as explicit global input to a step. 

```

Comment (philtweir) [reply docs/workflows.md:112]: As a side-note, this is interesting context: https://github.com/python/typing/discussions/1349

Comment (philtweir) [reply docs/workflows.md]: Updated accordingly.

**Pull request merged**